src/modules/layout/layoutReplaceDimension.js

```
@@ -6,6 +6,10 @@ export const layoutReplaceDimension = (layout, dimensionId, items) => {
         axisHasDimension(layout[a], dimensionId)
     )
 
+    if (!axisName) {
```

I'm not sure that just returning a copy of the original object is the right thing to do. If something is wrong (i.e., the dimension isn't found in any of the axes), then the consumer of this code ought to be notified? Otherwise it just carries on, as though everything had succeeded. Which could lead to problems later. I would rather throw an error. Returning undefined is also an option, though less informative.